### PR TITLE
add `analysis` schema for event-aligned Calcium activity

### PR DESCRIPTION
**Pull request by ttngu207**
First pass, could use some more testing, review, suggestions

### CODE REVIEW

**Comment by CBroz1** [comment]
Thanks for the updates here, @ttngu207. Looks good. I'll pull from your fork and attempt running soon. When I have a notebook draft, I'll push changes there.

File: workflow_calcium_imaging/analysis.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+        trace_keys, activity_traces = (imaging.Activity.Trace & key).fetch('KEY', 'activity_trace', order_by='mask')
+        activity_traces = np.hstack(activity_traces)
+
+        aligned_trial_activities = []
+        for _, r in trialized_event_times.iterrows():
+            if r.event is None or np.isnan(r.event):
+                continue
+            alignment_start_time = r.event - min_limit
+            alignment_end_time = r.event + max_limit
+            roi_aligned_activities = activity_traces[:, (alignment_start_time <= frame_timestamps)
```
Comment (CBroz1) [inline workflow_calcium_imaging/analysis.py:70]: `activity_traces` comes back for me as one-dimensional when following the notebook. Any insight as to why that might be? I'll continue to debug
```
IndexError: too many indices for array: array is 1-dimensional, but 2 were indexed
```

File: workflow_calcium_imaging/analysis.py [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+                                                        & (frame_timestamps < alignment_end_time)]
+
+            aligned_trial_activities.extend([{**key, **r.trial_key, **trace_key, 'aligned_trace': aligned_trace}
+                                             for trace_key, aligned_trace in zip(trace_keys, roi_aligned_activities)])
+
+        self.insert1({**key, 'aligned_timestamps': np.linspace(
+            -min_limit, max_limit, len(aligned_trial_activities[0]['aligned_trace']))})
+        self.AlignedTrialSpikes.insert(aligned_trial_activities)
+
+    def plot_aligned_activities(self, key, roi, axs=None):
```
Comment (CBroz1) [inline workflow_calcium_imaging/analysis.py:80]: In the notebook, this is called with `unit=2` - Should that be `roi=2`? If I make that adjustment, L85 below gives me the following error
```
DataJointError: Attribute `aligned_trace` not found.
```

Comment (CBroz1) [reply workflow_calcium_imaging/analysis.py:80]: If I bring everything into the parent namespace, I get `TypeError: Invalid shape (2822,) for image data`

```python
import datetime
key = {'subject': 'subject3', 
       'session_datetime': datetime.datetime(2021, 10, 25, 13, 6, 40), 
       'scan_id': 0, 'paramset_idx': 0, 'curation_id': 1, 
       'extraction_method': 'suite2p_deconvolution', 
       'alignment_name': 'center_button', 'trial_condition': 'ctrl_center_button'}
trial_ids, aligned_spikes= (analysis.ActivityAlignment.AlignedTrialSpikes 
                            & key & {'mask': 2}
                           ).fetch('trial_id', 'aligned_trace', order_by='trial_id')
import numpy as np
aligned_spikes = np.hstack(aligned_spikes)
import matplotlib.pyplot as plt
fig = None
fig, (ax0, ax1) = plt.subplots(2, 1, figsize=(12, 8))
ax0.imshow(aligned_spikes, cmap='gray', interpolation='nearest')
ax1.plot(aligned_timestamps, np.nanmean(aligned_spikes))
```

File: workflow_calcium_imaging/analysis.py [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+        import matplotlib.pyplot as plt
+
+        fig = None
+        if axs is None:
+            fig, (ax0, ax1) = plt.subplots(2, 1, figsize=(12, 8))
+        else:
+            ax0, ax1 = axs
+
+        aligned_timestamps = (self & key).fetch1('aligned_trace')
```
Comment (CBroz1) [inline workflow_calcium_imaging/analysis.py]:
```suggestion
        aligned_timestamps = (self & key).fetch1('aligned_timestamps')
```

File: workflow_calcium_imaging/analysis.py [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+        trial_ids, aligned_spikes = (self.AlignedTrialSpikes
+                                     & key & {'mask': roi}).fetch('trial_id', 'aligned_trace', order_by='trial_id')
+
+        aligned_spikes = np.hstack(aligned_spikes)
+
+        ax0.imshow(aligned_spikes, cmap='gray', interpolation='nearest')
```
Comment (CBroz1) [inline workflow_calcium_imaging/analysis.py:96]: `aligned_spike` was coming back for me as a 1d array, so I changed to 
```suggestion
        ax0.plot(aligned_spikes)
```
But errors persist - `aligned_spikes` and `aligned_timestamps` differ in size

File: notebooks/py_scripts/07-downstream-analysis-optional.py [hunk 5]
```diff
@@ -0,0 +1,204 @@
...
+# Just like the element itself, we can run computations with `populate()`
+# -
+
+analysis.ActivityAlignment.populate(display_progress=True)
+
+analysis.ActivityAlignment()
+
+# The `AlignedTrialSpikes` part table captures aligned traces fore each alignment and trial condition specified in the master table.
+
+analysis.ActivityAlignment.AlignedTrialSpikes()
```
Comment (kabilar) [inline notebooks/py_scripts/07-downstream-analysis-optional.py:174]:
```suggestion
# The `AlignedTrialActivity` part table captures aligned traces fore each alignment and trial condition specified in the master table.

analysis.ActivityAlignment.AlignedTrialActivity()
```

**Review by kabilar — APPROVED** [review]
Thanks @ttngu207 and @CBroz1!  This is great work.  Just one minor suggestion above.

File: notebooks/07-downstream-analysis-optional.ipynb [hunk 6]
```diff
@@ -0,0 +1,2955 @@
...
+   "source": [
+    "analysis.ActivityAlignment()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "e014bece-ce78-41fe-9f69-b05beee1d307",
+   "metadata": {},
+   "source": [
+    "The `AlignedTrialSpikes` part table captures aligned traces fore each alignment and trial condition specified in the master table."
```
Comment (kabilar) [inline notebooks/07-downstream-analysis-optional.ipynb:2391]:
```suggestion
    "The `AlignedTrialActivity` part table captures aligned traces fore each alignment and trial condition specified in the master table."
```

File: notebooks/07-downstream-analysis-optional.ipynb [hunk 7]
```diff
@@ -0,0 +1,2955 @@
...
+       " (Total: 126084)"
+      ]
+     },
+     "execution_count": 6,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "analysis.ActivityAlignment.AlignedTrialSpikes()"
```
Comment (kabilar) [inline notebooks/07-downstream-analysis-optional.ipynb:2632]:
```suggestion
    "analysis.ActivityAlignment.AlignedTrialActivity()"
```

**Review by kabilar — APPROVED** [review]

**Pull request merged**